test/index.js

```diff
@@ -1261,6 +1261,7 @@ test('extend', function() {
   eq(Z.extend(joinWith(''), ['x', 'y']), ['xy', 'y']);
   eq(Z.extend(joinWith(''), ['x', 'y', 'z']), ['xyz', 'yz', 'z']);
   eq(Z.extend(function(id) { return Z.reduce(add, 1, id); }, Identity(42)), Identity(43));
+  eq(Z.extend(function(f) { return f([1, 2]) + '2'; }, function(xs) { return xs.length; })([3, 4]), '42');
```

Comment:
Let's replace the second function expression with `length` (defined near the top of the file).

Reply:
Actually, the expression above relies on type coercion. Let's avoid this. I'd also like the expression to demonstrate the order of arguments to `concat`. I suggest:

```javascript
> Z.extend(function(f) { return f([3, 4]); }, Z.reverse)([1, 2])
[4, 3, 2, 1]
```

Further improvements much appreciated. :)

Reply:
I can't think of any at the moment.